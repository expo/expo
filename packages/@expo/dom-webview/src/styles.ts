import { StyleSheet } from 'react-native';

export const webviewStyles = StyleSheet.create({
  container: {
    flex: 1,
    overflow: 'hidden',
  },
  webView: {
    backgroundColor: '#ffffff',
  },
});
