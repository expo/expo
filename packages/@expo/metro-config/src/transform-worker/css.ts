export function pathToHtmlSafeName(path: string) {
  return path.replace(/[^a-zA-Z0-9_]/g, '_');
}

export function getHotReplaceTemplate(id: string) {
  // In dev mode, we need to replace the style tag instead of appending it
  // use the path as the expo-css-hmr attribute to find the style tag
  // to replace.
  const attr = JSON.stringify(pathToHtmlSafeName(id));
  return `style.setAttribute('data-expo-css-hmr', ${attr});
  const previousStyle = document.querySelector('[data-expo-css-hmr=${attr}]');
  if (previousStyle) {
    previousStyle.parentNode.removeChild(previousStyle);
  }`;
}

export function wrapDevelopmentCSS(props: { src: string; filename: string }) {
  const withBackTicksEscaped = escapeBackticksAndOctals(props.src);
  return `(() => {
  if (typeof document === 'undefined') {
    return
  }
  const head = document.head || document.getElementsByTagName('head')[0];
  const style = document.createElement('style');
  ${getHotReplaceTemplate(props.filename)}
  style.setAttribute('data-expo-loader', 'css');
  head.appendChild(style);
  const css = \`${withBackTicksEscaped}\`;
  if (style.styleSheet){
    style.styleSheet.cssText = css;
  } else {
    style.appendChild(document.createTextNode(css));
  }
})();`;
}

export function escapeBackticksAndOctals(str: string) {
  if (typeof str !== 'string') {
    return '';
  }

  return str
    .replace(/\\/g, '\\\\')
    .replace(/`/g, '\\`')
    .replace(/[\0-\7]/g, (match) => `\\0${match.charCodeAt(0).toString(8)}`);
}
