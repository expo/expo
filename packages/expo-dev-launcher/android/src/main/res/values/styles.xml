<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:tools="http://schemas.android.com/tools">
  <style name="Theme.DevLauncher.LauncherActivity" parent="Theme.AppCompat.Light.NoActionBar">
    <item name="android:navigationBarColor">@android:color/transparent</item>
    <item name="android:statusBarColor">@android:color/transparent</item>
    <item name="android:windowLightStatusBar">true</item>
    <item name="android:enforceNavigationBarContrast" tools:targetApi="q">false</item>
    <item name="android:windowLightNavigationBar" tools:targetApi="o_mr1">true</item>

    <item name="android:windowContentOverlay">@null</item>
    <item name="android:windowNoTitle">true</item>
    <item name="android:windowIsFloating">false</item>
    <item name="android:backgroundDimEnabled">false</item>
    <item name="android:windowAnimationStyle">@null</item>
  </style>

  <style name="Theme.DevLauncher.ErrorActivity" parent="Theme.AppCompat.Light.NoActionBar">
    <item name="android:navigationBarColor">@android:color/transparent</item>
    <item name="android:statusBarColor">@android:color/transparent</item>
    <item name="android:windowLightStatusBar">true</item>
    <item name="android:enforceNavigationBarContrast" tools:targetApi="q">false</item>
    <item name="android:windowLightNavigationBar" tools:targetApi="o_mr1">true</item>

    <item name="colorPrimary">@color/dev_launcher_primary</item>
    <item name="colorPrimaryDark">@color/dev_launcher_colorPrimaryDark</item>
    <item name="colorAccent">@color/dev_launcher_colorAccentDark</item>
    <item name="android:windowBackground">@color/dev_launcher_backgroundColor</item>
  </style>
</resources>
