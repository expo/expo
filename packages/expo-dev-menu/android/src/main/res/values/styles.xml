<?xml version="1.0" encoding="utf-8"?>
<resources xmlns:tools="http://schemas.android.com/tools">

  <style name="Theme.AppCompat.Transparent.NoActionBar" parent="Theme.AppCompat.Light.NoActionBar">
    <item name="android:windowIsTranslucent">true</item>
    <item name="android:windowBackground">@android:color/transparent</item>
    <item name="android:windowContentOverlay">@null</item>
    <item name="android:windowNoTitle">true</item>
    <item name="android:windowIsFloating">false</item>
    <item name="android:backgroundDimEnabled">false</item>
    <item name="android:windowAnimationStyle">@null</item>
    <item name="android:enforceNavigationBarContrast" tools:targetApi="q">true</item>
    <item name="android:navigationBarColor">@android:color/transparent</item>
    <item name="android:statusBarColor">@android:color/transparent</item>
    <item name="android:windowDrawsSystemBarBackgrounds">true</item>
  </style>

</resources>
