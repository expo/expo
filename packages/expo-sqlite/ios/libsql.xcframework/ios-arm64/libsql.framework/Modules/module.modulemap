framework module libsql {
  umbrella header "libsql-umbrella.h"
  export *
  module * { export * }
}
