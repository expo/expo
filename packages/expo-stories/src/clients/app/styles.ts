import { lightTheme, shadows, spacing } from '@expo/styleguide-native';
import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  flexContainer: {
    flex: 1,
    backgroundColor: lightTheme.background.default,
  },
  storyRow: {},
  storyButtonRow: {
    padding: spacing[4],
  },
  storyTitle: {
    paddingVertical: spacing[2],
    paddingHorizontal: spacing[3],
    marginTop: spacing[4],
    marginBottom: spacing[1],
    fontSize: 20,
    fontWeight: '700',
  },
  storyButton: {
    borderRadius: 4,
    paddingVertical: spacing[4],
    marginVertical: spacing[2],
    backgroundColor: lightTheme.button.primary.background,
    ...shadows.button,
  },
  storyButtonText: {
    fontSize: 18,
    fontWeight: '700',
    color: lightTheme.button.primary.foreground,
    textAlign: 'center',
  },
  refreshButton: {
    position: 'absolute',
    padding: spacing[3],
    bottom: spacing[6],
    left: 0,
    right: 0,
  },
  refreshLoader: {
    position: 'absolute',
    right: spacing[4],
    bottom: 0,
    top: 0,
  },
  loadingContainer: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'center',
    alignItems: 'center',
  },
  listContainer: {
    paddingBottom: spacing[24] + spacing[6],
  },
  seeSelectionButtonContainer: {
    position: 'absolute',
    bottom: spacing[24],
    left: 0,
    right: 0,
    paddingHorizontal: spacing[6],
  },
  sectionHeaderContainer: {
    paddingTop: spacing[6],
    paddingBottom: spacing[2],
    paddingHorizontal: spacing[3],
  },
  sectionHeader: {
    fontSize: 24,
    fontWeight: '700',
    color: lightTheme.text.default,
  },
});
