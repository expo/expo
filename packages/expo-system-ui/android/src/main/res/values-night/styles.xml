<?xml version="1.0" encoding="utf-8"?>
<resources>
    <style name="Theme.EdgeToEdge.Base" parent="Theme.AppCompat.DayNight.NoActionBar">
        <item name="windowLightSystemBars">false</item>
    </style>
</resources>
