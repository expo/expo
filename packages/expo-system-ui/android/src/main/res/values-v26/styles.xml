<?xml version="1.0" encoding="utf-8"?>
<resources>
    <style name="Theme.EdgeToEdge" parent="Theme.EdgeToEdge.Common">
        <item name="android:navigationBarColor">@color/systemBarLightScrim</item>
    </style>
</resources>
