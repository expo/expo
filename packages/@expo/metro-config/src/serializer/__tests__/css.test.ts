import { serializeOptimizeAsync } from '../fork/__tests__/serializer-test-utils';

jest.mock('../exportHermes', () => {
  return {
    buildHermesBundleAsync: jest.fn(({ code, map }) => ({
      hbc: code,
      sourcemap: map,
    })),
  };
});

const originalWarn = console.warn;

beforeEach(() => {
  console.warn = jest.fn(originalWarn);
});

afterAll(() => {
  console.warn = originalWarn;
});

it(`supports global CSS files`, async () => {
  const [, artifacts] = await serializeOptimizeAsync({
    'index.js': `
          import './styles.css';
        `,
    'styles.css': `.container {
    color: blue;
    }
        `,
  });

  expect(artifacts[1].source).toBe('.container {\n  color: #00f;\n}\n');
});

it(`supports global CSS files with local imports`, async () => {
  const [, artifacts] = await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.css';
        `,
      'styles.css': `
        @import './other.css';
    .container {
    color: dodgerblue;
    }
        `,
      'other.css': `

    .second {
    color: red;
    }
        `,
    },
    {
      minify: true,
    }
  );
  expect(artifacts.length).toBe(3);

  expect(artifacts[1].source).toMatch('red');
  expect(artifacts[2].source).toMatch('#1e90ff');
});

it(`supports global CSS files with URL imports`, async () => {
  const [, artifacts] = await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.css';
        `,
      'styles.css': `
        @import url('https://example.com/other.css');
        .container {
          color: dodgerblue;
        }
        `,
    },
    {
      minify: true,
    }
  );
  expect(artifacts.length).toBe(3);

  expect(artifacts[1].source).toMatch(
    '<link rel="stylesheet" href="https://example.com/other.css">'
  );
  expect(artifacts[2].source).toMatch('#1e90ff');
});

it(`supports url for style attributes`, async () => {
  const [, artifacts] = await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.css';
        `,
      'styles.css': `        
        .container {
          background: url('https://example.com/image.png');
        }
        `,
    },
    {
      minify: true,
    }
  );
  expect(artifacts.length).toBe(2);
  expect(artifacts[1].source).toMatch(
    '.container{background:url("https://example.com/image.png")}'
  );
});

it(`supports url with abstract imports for style attributes`, async () => {
  const [, artifacts] = await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.css';
        `,
      'styles.css': `        
        .other {
          background: url("data:image/svg+xml;charset=utf8,...");
        }
        .container {
          background-image: url(/icon-mask.svg);
          stroke: url(#edge-gradient);
        }
        `,
    },
    {
      minify: true,
    }
  );
  expect(artifacts.length).toBe(2);
  expect(artifacts[1].source).toMatch(
    '.other{background:url("data:image/svg+xml;charset=utf8,...")}.container{stroke:url("#edge-gradient");background-image:url("/icon-mask.svg")}'
  );
});

it(`supports url with abstract imports for style attributes in CSS module`, async () => {
  const [, artifacts] = await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.module.css';
        `,
      'styles.module.css': `        
        .appIcon {
            mask: url("data:xxx") center/100% 100% no-repeat;
        }
        `,
    },
    {
      minify: true,
      dev: true,
    }
  );
  expect(artifacts.length).toBe(2);
  // Ensure the HMR code contains the updates.
  expect(artifacts[0].source).toMatch('mask:url(\\"data:xxx\\")');
  expect(artifacts[1].source).toMatch(
    '.EcQGha_appIcon{mask:url("data:xxx") 50%/100% 100% no-repeat}'
  );
});

it(`asserts that local imports in attributes are not yet supported`, async () => {
  await serializeOptimizeAsync(
    {
      'index.js': `
          import './styles.css';
        `,
      'styles.css': `        
        .container {
          background: url('./image.png');
        }
        `,
    },
    {
      minify: true,
    }
  );

  expect(console.warn).toHaveBeenCalledTimes(1);
});

describe('css modules', () => {
  // TODO: We may want to roll this functionality back in the future since it introduces unscoped CSS in CSS modules.
  it(`supports local imports for style attributes in CSS modules`, async () => {
    const [, artifacts] = await serializeOptimizeAsync(
      {
        'index.js': `
            import styles from './styles.module.css';
            console.log(styles);
          `,
        'styles.module.css': `
          @import './other.css';
      .container {
      color: dodgerblue;
      }
          `,
        'other.css': `

      .second {
      color: red;
      }
          `,
      },
      {
        minify: true,
      }
    );
    expect(artifacts.length).toBe(3);

    expect(artifacts[1].source).toMatch('red');
    expect(artifacts[2].source).toMatch('#1e90ff');
  });

  it(`supports remote imports for style attributes in CSS modules`, async () => {
    const [, artifacts] = await serializeOptimizeAsync(
      {
        'index.js': `
            import styles from './styles.module.css';
            console.log(styles);
          `,
        'styles.module.css': `
          @import url('https://example.com/image.png');
      .container {
      color: dodgerblue;
      }
          `,
      },
      {
        minify: true,
      }
    );
    expect(artifacts.length).toBe(3);

    expect(artifacts[1].source).toMatch(
      '<link rel="stylesheet" href="https://example.com/image.png">'
    );
    expect(artifacts[2].source).toMatch('#1e90ff');
  });
});
