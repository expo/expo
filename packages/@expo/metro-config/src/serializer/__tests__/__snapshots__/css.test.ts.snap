// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`asserts that local imports in attributes are not yet supported 1`] = `
"Importing local resources in CSS is not supported yet. (/app/styles.css:3:27):
  1 |         
  2 |         .container {
> 3 |           background: url('./image.png');
    |                           ^
  4 |         }
  5 |         "
`;
